webapp/src/main/java/com/box/l10n/mojito/service/assetExtraction/AssetExtractionService.java

```diff
@@ -196,77 +200,136 @@ public class AssetExtractionService {
         Asset asset = getUndeletedAsset(assetContent.getAsset());
 
-        MultiBranchState stateForNewContent = convertAssetContentToMultiBranchState(assetContent, filterConfigIdOverride, filterOptions, currentTask);
-        CreateTextUnitsResult createdTextUnitsResult = createTextUnitsForNewContent(assetContent, stateForNewContent, currentTask);
+        MultiBranchState stateForNewContent = convertAssetContentToMultiBranchState(assetContent,
+                                                                                    filterConfigIdOverride,
+                                                                                    filterOptions,
+                                                                                    currentTask);
+        CreateTextUnitsResult createdTextUnitsResult = createTextUnitsForNewContent(assetContent,
+                                                                                    stateForNewContent,
+                                                                                    currentTask);
 
         updateBranchAssetExtraction(assetContent, createdTextUnitsResult.getUpdatedState(), filterOptions, currentTask);
         updateLastSuccessfulAssetExtraction(asset, createdTextUnitsResult.getUpdatedState(), currentTask);
+        updatePushRun(asset, createdTextUnitsResult.getUpdatedState(), pushRunId, currentTask);
         performLeveraging(createdTextUnitsResult.getLeveragingMatches(), currentTask);
 
         logger.info("Done processing asset content id: {}", assetContentId);
         return new PollableFutureTaskResult<>(asset);
     }
 
     @Pollable(message = "Updating merged asset text units")
-    void updateLastSuccessfulAssetExtraction(Asset asset, MultiBranchState currentState, @ParentTask PollableTask currentTask) {
+    void updateLastSuccessfulAssetExtraction(Asset asset,
+                                                         MultiBranchState currentState,
+                                                         @ParentTask PollableTask currentTask) {
         logger.trace("Make sure we have a last successful extraction in the Asset (legacy support edge case)");
         AssetExtraction lastSuccessfulAssetExtraction = getOrCreateLastSuccessfulAssetExtraction(asset);
-        MultiBranchState lastSuccessfulMultiBranchState = updateAssetExtractionWithState(lastSuccessfulAssetExtraction.getId(), currentState, AssetContentMd5s.of());
+        updateAssetExtractionWithState(lastSuccessfulAssetExtraction.getId(),
+                                       currentState,
+                                       AssetContentMd5s.of());
+    }
+
+    @Pollable(message = "Updating push run")
+    void updatePushRun(Asset asset,
+                       MultiBranchState multiBranchState,
+                       Long pushRunId,
+                       @ParentTask PollableTask parentTask) {
+
+        if (pushRunId == null || pushRunId == 0) {
```

Comment:
why `pushRunId == 0`?

Reply:
AutoIncrement in SQL starts at 1 and thus 0 is always guaranteed to be an incorrect value - likely passed in for a badly initialized param (same as null effectively).